phoenix-core/src/it/java/org/apache/phoenix/mapreduce/VerifyReplicationToolIT.java

```diff
@@ -0,0 +1,323 @@
+        assertEquals(10, countUsers(targetTableName, sourceAndTargetTenant));
+        assertEquals(10, countUsers(targetTableName, targetOnlyTenant));
+        assertEquals(0, countUsers(targetTableName, sourceOnlyTenant));
+    }
+
+    private void upsertData(PreparedStatement stmt, String tenantId) throws SQLException {
+        String userId = "user" + userNum++;
+        int age = RANDOM.nextInt(99) + 1;
+        upsertData(stmt, tenantId, userId, age);
+    }
+
+    private void upsertData(PreparedStatement stmt, String tenantId, String userId, int age)
+            throws SQLException {
+        int i = 1;
+        stmt.setString(i++, tenantId);
+        stmt.setString(i++, userId);
+        stmt.setInt(i++, age);
+        stmt.execute();
+    }
+
+    private void upsertSelectData(String tableName, String tenantId, int age, int limit)
+            throws SQLException {
+        String sql = String.format(UPSERT_SELECT_USERS, tableName, age, tableName, limit);
+        Connection conn = DriverManager.getConnection(getUrl());
+        PreparedStatement ps = conn.prepareStatement(sql);
+        ps.setString(1, tenantId);
+        ps.execute();
+        conn.commit();
+        assertEquals(10, countUsers(tableName, tenantId));
+    }
+
+    private int countUsers(String tableName, String tenantId) throws SQLException {
+        Connection conn = DriverManager.getConnection(getUrl());
+        PreparedStatement ps =
+                conn.prepareStatement(
+                        String.format("SELECT COUNT(*) FROM %s WHERE TENANT_ID = ?", tableName));
+        ps.setString(1, tenantId);
+        ResultSet rs = ps.executeQuery();
+        rs.next();
+        return rs.getInt(1);
+    }
+
+    private void cleanupPreviousJobOutput(Job job) throws IOException {
+        Path outputDir =
+            new Path(job.getConfiguration().get("mapreduce.output.fileoutputformat.outputdir"));
+        FileSystem fs = FileSystem.get(getUtility().getConfiguration());
+        if (fs.exists(outputDir)) {
+            fs.delete(outputDir, true);
+        }
+    }
+
+    private void split(String tableName, int targetRegions) throws Exception {
+        TableName table = TableName.valueOf(tableName);
+        HBaseAdmin admin = driver.getConnectionQueryServices(getUrl(), TEST_PROPERTIES).getAdmin();
+        MiniHBaseCluster cluster = getUtility().getHBaseCluster();
+        HMaster master = cluster.getMaster();
+        List<HRegionInfo> regions = admin.getTableRegions(table);
+        int numRegions = regions.size();
+        // split the last region in the table until we have the target number of regions
+        while (numRegions < targetRegions) {
+            HRegionInfo region = regions.get(numRegions - 1);
+            ServerName serverName =
+                    master.getAssignmentManager().getRegionStates()
+                            .getRegionServerOfRegion(region);
+            byte[] splitPoint =
```

Comment:
I am not sure if this is working as intended. You want the data to be on different regions, right, so that multiple map tasks can be spawned?
The first region has start key `''` and the split will give it the end key as `\x0` and the same will go on for all the splits. Hence all of your data will be in single region only. When I put a break point in `PhoenixInputFormat#getSplits()`, the query plan just returns 1 split for the tenant.

Reply:
@karanmehta93 : This is part of the code which @aaraujo originally wrote. I didn't get a chance to dive deeper into it yet, I am still adding more test cases. You should see the functionality change in the next update. Thanks!

Reply:
Karan is right, you will have all the rows on a single region, and the remaining regions without any rows. 
You can use ByteUtil.concat(tenantId, QueryConstants.SEPARATOR_BYTE_ARRAY, userId) to split on the rowkey of a particular row to ensure that rows will be on multiple regions.